hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bucket/BucketIdentifier.java

```diff
@@ -115,6 +115,13 @@ private static String newBucketFileIdFixedSuffix(int bucketId) {
     return newBucketFileIdFixedSuffix(bucketIdStr(bucketId));
   }
 
+  /**
+   * Generate a new file id for NBCC mode, file id is fixed for each bucket with format: "{bucket_id}-0000-0000-0000-000000000000-0"
+   */
+  public static String newBucketFileIdForNBCC(int bucketId) {
+    return FSUtils.createNewFileId(newBucketFileIdFixedSuffix(bucketId), 0);
```

Comment:
Is the change compatible with existing behavior?

Reply:
> Is the change compatible with existing behavior?

The bucket with existing data will be written with the existing file name. Only the newly added bucket will use the new name, which does not cause compatibility problems